src/wallet/asyncrpcoperation_saplingmigration.cpp

```diff
@@ -201,27 +203,10 @@ libzcash::SaplingPaymentAddress AsyncRPCOperation_saplingmigration::getMigration
         assert(saplingAddress != nullptr); // This is checked in init.cpp
         return *saplingAddress;
     }
-    // Derive the address for Sapling account 0
-    auto m = libzcash::SaplingExtendedSpendingKey::Master(seed);
-    uint32_t bip44CoinType = Params().BIP44CoinType();
 
-    // We use a fixed keypath scheme of m/32'/coin_type'/account'
-    // Derive m/32'
-    auto m_32h = m.Derive(32 | ZIP32_HARDENED_KEY_LIMIT);
-    // Derive m/32'/coin_type'
-    auto m_32h_cth = m_32h.Derive(bip44CoinType | ZIP32_HARDENED_KEY_LIMIT);
-
-    // Derive m/32'/coin_type'/0'
-    libzcash::SaplingExtendedSpendingKey xsk = m_32h_cth.Derive(0 | ZIP32_HARDENED_KEY_LIMIT);
-
-    libzcash::SaplingPaymentAddress toAddress = xsk.DefaultAddress();
-
-    if (!HaveSpendingKeyForPaymentAddress(pwalletMain)(toAddress)) {
-        // Sapling account 0 must be the first address returned by GenerateNewSaplingZKey
-        assert(pwalletMain->GenerateNewSaplingZKey() == toAddress);
-    }
-
-    return toAddress;
+    // TODO: move off of legacy addresses.
+    auto generatedKey = pwalletMain->GenerateLegacySaplingZKey(0);
```

Comment:
(Haven't reviewed it yet, so) what does this API do if the address already exists? Previously we derived the expected address, and then generated a new key in the wallet if it didn't already exist, to avoid any potential for unnecessarily overwriting existing key material in the `wallet.dat`.

Reply:
If the key already exists in the wallet, it just returns it.